make_count/make_count.tex

```diff
@@ -0,0 +1,293 @@
+    \end{itemize}
+\end{itemize}
+
+\textbf{Postconditions:} a \texttt{Transformation} must be returned (i.e. if a \texttt{Transformation} cannot be returned successfully, a runtime error should be returned)
+
+\begin{lstlisting}[language = Python, escapechar=|]
+def MakeCount(TIA, TO):
+
+    input_domain = VectorDomain(AllDomain(TIA))
+    output_domain = AllDomain(TO)
+    input_metric = SymmetricDistance()
+    output_metric = AbsoluteDistance(TO)
+    
+    # give the Transformation the following properties
+    max_value = get_max_consecutive_int(TO) 
+    def function(data: Vec<TIA>) -> TO:|\label{line:TO-output}|
+        try:
+            return exact_int_cast(len(data), TO)
+        except FailedCast:
+            return max_value |\label{line:except-return}|
+    def stability_relation(din:u32, dout:TO) -> bool: |\label{line:din-dout-type}|
+        return 1 * inf_cast(din,TO) <= dout |\label{line:stability-relation}|
+
+    # now, return the Transformation
+    return Transformation(input_domain,output_domain,function,input_metric,output_metric,stability_relation)
+
+\end{lstlisting}
+
+\section{Proofs for the pseudocode}
+
+\begin{theorem}
+\label{thrm:domain-and-stability}
+    For every setting of the input parameters \texttt{TIA}, \texttt{TO} for \texttt{MakeCount} such that the given preconditions hold, the \texttt{Transformation} returned by \texttt{MakeCount} has the following properties:
+    \begin{enumerate}
+        \item \textup{(Appropriate output domain).} For every vector \texttt{v} in the \texttt{input\_domain}, \texttt{function(v)} is in the \texttt{output\_domain}.
+        
+        \item \textup{(Domain-metric compatibility).} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}, and likewise \texttt{output\_domain} matches one of the possible domains listed in the definition of \texttt{output\_metric}.
+        
+        \item \textup{(Stability guarantee).} For every input \texttt{u}, \texttt{v} drawn from the \texttt{input\_domain} and for every pair $(\din, \dout)$, where $\din$ is of type \texttt{u32} and $\dout$ is of type \texttt{TO} (see line \ref{line:din-dout-type} of the pseudocode), if \texttt{u}, \texttt{v} are $\din$-close under the \texttt{input\_metric} and $\texttt{stability\_relation(din, dout) = True}$, then \texttt{function(u)}, \texttt{function(v)} are $\dout$-close under the \texttt{output\_metric}.
+    \end{enumerate}
+\end{theorem}
+
+\begin{proof} \textbf{(Part 1 -- appropriate output domain).}
+In section \ref{sec:python-pseudocode}, we see that any value of type \texttt{TO} is in the \texttt{output\_domain}, and in line \ref{line:TO-output} of the Python-style pseudocode, we see that the \texttt{function} is always guaranteed to return a value of type \texttt{TO}. Therefore, since our output domain is any value of type \texttt{TO}, we see that $\function$ has the appropriate output domain \texttt{output\_domain}.

```

Comment:
Rather than "In section ..." I'd say "In line 4 of the pseudocode, we have output_domain=AllDomain(TO), so every value of type TO..."  (try to always use "every" instead of "any" - the latter is ambiguous as a quantifier)